Assignment4/jdlgc-a180369/task07.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/10bZVZLj1O3vIV6-4mylFF0U9xHmKuD2I
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+"""First let's read the RDF file"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+ns = Namespace("http://somewhere#")
+from rdflib.plugins.sparql import prepareQuery
+
+print("RDFLib:")
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
```

Comment:
This does not retrieve the subclasses of subclasses of living thing. Hint: use a recursive function